src/pages/index.js

```diff
@@ -20,42 +33,75 @@ function openPreviewModal(cardData) {
   imageModal.open(cardData);
 }
 
+// Confirm Delete Modal
+const deleteCardModal = new PopupWithConfirm("#card-delete-modal");
+deleteCardModal.setEventListeners();
+
+function openDeleteCardModal(cardId, cardElement) {
+  deleteCardModal.setSubmitAction(() => {
+    deleteCardModal.setIsLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove(); // Remove the card from the DOM immediately after successful deletion
+        deleteCardModal.close();
+      })
+      .catch((err) => {
+        console.error(`Error deleting card: ${err}`);
+        alert("Failed to delete the card. Please try again."); // Optional user feedback
+      })
+      .finally(() => deleteCardModal.setIsLoading(false));
+  });
+  deleteCardModal.open();
+}
+
 function createCard(data) {
-  const card = new Card(data, "#card-template", openPreviewModal);
+  const card = new Card(
+    data,
+    "#card-template",
+    openPreviewModal,
+    (cardId, isLiked) => {
+      const apiCall = isLiked ? api.removeLike(cardId) : api.addLike(cardId);
+      apiCall.then((updatedCard) => {
+        card.updateLikes(updatedCard.likes); // Updates likes dynamically in the UI
+      });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**